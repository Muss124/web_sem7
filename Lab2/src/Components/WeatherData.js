import React from "react"
import store from "../store.js"

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		console.log("here");
		console.log(WeatherData)
		return (
			<div class="WeatherData">
				<div class="WeatherTop">
					<h2>{WeatherData["name"]}</h2>
					<img src={"http://openweathermap.org/img/wn/" + WeatherData["weather"][0]["icon"] + ".png"} alt=""></img>
					<h1>{WeatherData["main"]["temp"]} &#8451;</h1>
					<button type="submit" class="btn btn-secondary" onClick={() => {
						console.log("REMOVE");
						store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData["name"] });
					}} >&#10006;</button>
				</div>
				<div class="WeatherMain">
					<div class="WeatherMainEl">
						<div class="WeatherParam">Weather</div>
						<div class="WeatherValue">{WeatherData["weather"][0]["description"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Humidity (%)</div>
						<div class="WeatherValue">{WeatherData["main"]["humidity"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Pressure (hPa)</div>
						<div class="WeatherValue">{WeatherData["main"]["pressure"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Wind speed (meter/sec)</div>
						<div class="WeatherValue">{WeatherData["wind"]["speed"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Longitude </div>
						<div class="WeatherValue">{WeatherData["coord"]["lon"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Latitude</div>
						<div class="WeatherValue">{WeatherData["coord"]["lat"]}</div>
					</div>
				</div>
			</div>
		);
	}
}