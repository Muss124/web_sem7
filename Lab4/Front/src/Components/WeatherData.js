import React from "react"
import store from "../store.js"

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		console.log("here");
		console.log(WeatherData)
		return (
			<div class="WeatherData">
				<div class="WeatherTop">
					<h2>{WeatherData.CityName}</h2>
					<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
					<h1>{WeatherData.Temperature} &#8451;</h1>
					<button type="submit" class="btn btn-secondary" onClick={() => {
						console.log("REMOVE");
						store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
					}} >&#10006;</button>
				</div>
				<div class="WeatherMain">
					<div class="WeatherMainEl">
						<div class="WeatherParam">Weather</div>
						<div class="WeatherValue">{WeatherData.Weather}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Humidity (%)</div>
						<div class="WeatherValue">{WeatherData.Humidity}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Pressure (hPa)</div>
						<div class="WeatherValue">{WeatherData.Pressure}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Wind speed (meter/sec)</div>
						<div class="WeatherValue">{WeatherData.Wind}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Longitude </div>
						<div class="WeatherValue">{WeatherData.Longitude}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Latitude</div>
						<div class="WeatherValue">{WeatherData.Latitude}</div>
					</div>
				</div>
			</div>
		);
	}
}