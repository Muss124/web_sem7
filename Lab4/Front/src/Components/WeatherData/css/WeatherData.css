.WeatherMain {
    margin-top   : auto;
    margin-bottom: auto;
}

.TopCityWeather .WeatherData .WeatherTop h1 {
    width     : 80%;
    text-align: right;
    font-size : 50px;
}

.WeatherTop {
    display: inline-block;
}

.WeatherMainEl {
    display         : flex;
    justify-content : space-between;
    background-color: #999;
    margin          : 5px;
    padding         : 5px;
}

.WeatherTop {
    display        : flex;
    justify-content: flex-start;
}


.WeatherParam {
    font-weight: bold;
}