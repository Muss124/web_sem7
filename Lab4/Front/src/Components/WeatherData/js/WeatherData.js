import React from "react"
import store from "../../../store"
import "../css/WeatherData.css"
import ServerDeleteCity from '../../../Actions/ServerDeleteCity'

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		//console.log("here");
		//console.log(WeatherData);
		//console.log(this.props.isMain);
		if (this.props.isMain) {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
						<button type="submit" className="btn btn-secondary" onClick={() => {
							console.log("REMOVE");
							//store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
							store.dispatch(ServerDeleteCity(WeatherData.CityName));
						}} >&#10006;</button>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}

	}
}