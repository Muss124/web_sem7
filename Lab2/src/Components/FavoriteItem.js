import React from 'react';
import {WeatherData} from "./WeatherData.js"

export class FavoriteItem extends React.Component {
	render() {
		
		if (this.props.value.loading) {
			return (
				<div>
					{this.props.value.city}
					<div class="spinner-border text-danger" role="status">
						<span class="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div class="FavoriteItem">
					<WeatherData value={this.props.value.data} />
				</div>
			);
		}
	}
}