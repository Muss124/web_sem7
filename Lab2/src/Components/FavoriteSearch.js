import React from "react"
import store from "../store.js"
import { itemsFetchData } from "../Actions/reques.js"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        // new shit 
        store.dispatch({ type: "FAVORITE_ADD", payload: { loading: true, city: value, data: [] } });
        store.dispatch(itemsFetchData(value));
        //store.dispatch({ type: "FAVORITE_DATA", payload: value });
        // old shit
        /*
        store.dispatch({ type: "FAVORITE_ADD_BEF", payload: { loading: true, city: value, data: [] } })
        var result = null;

        await fetch('http://api.openweathermap.org/data/2.5/weather?q=' + value + '&APPID=f8e5ebb3f762d1a58aaff7f643d7410b&units=metric')
            .then((res) => res.json()).then((res) => { result = res });
        console.log("recevied");
        console.log(result);
        if (result != null && result["cod"] === 200) {
            store.dispatch({ type: "FAVORITE_ADD_OK", payload: { loading: false, city: value, data: result } })
            //data.push(result);
            //window.localStorage.setItem("favorite", JSON.stringify(data));
            ////this.props.handler();
            ////var data = store.getState();
            ////this.setState({ loading: false, success: true, data: data });
        } else {
            alert("City doesn't founded(")
            store.dispatch({ type: "FAVORITE_ADD_ER", payload: value })
        }
        //window.localStorage.setItem('favorite', JSON.stringify(data));
        */
    }
    render() {
        return (
            <div class="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div class="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" class="btn btn-secondary btn-circle" >Add</button>
                    </form>
                </div>
            </div>
        );
    }
}