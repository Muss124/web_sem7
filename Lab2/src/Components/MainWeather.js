import React from "react"
import { WeatherData } from "./WeatherData.js"


export class MainWeather extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			loading: false
		}
	}
	componentDidMount() {
		this.checkGeo();
	}
	checkGeo() {
		if (navigator.geolocation === undefined) {
			this.badWeather();
		}
		else {
			navigator.geolocation.getCurrentPosition(this.goodWeather, this.badWeather)
		}
	}
	render() {
		if (this.state.loading) {
			return (
				<div class="spinner-border text-danger" role="status">
					<span class="sr-only">Loading...</span>
				</div>
			);
		}
		if (this.state.success) {
			var value = this.state.data;
			console.log(value);
			return (
				<div class="UpperWeather">
					<button type="button" class="btn btn-secondary" onClick={() => { this.setState({ loading: true }); this.checkGeo() }}>Refresh geolocation</button>
					<div class="TopCityWeather">
						<WeatherData value={value} />
					</div>
				</div>
			);
		}
		if (!this.state.access)
			return (<div><h1>Access error</h1></div>);

	}
	goodWeather = (position) => {
		this.setState({ loading: true, access: true });
		this.getWeather(position.coords.latitude, position.coords.longitude);
	}
	badWeather = () => {
		this.setState({ loading: true, access: false });
		this.getWeather(52.018082, 47.819569)
	}
	getWeather = async (latitude, longitude) => {
		var data = null;
		await fetch('http://api.openweathermap.org/data/2.5/weather?lat=' + latitude + '&lon=' + longitude + '&APPID=f8e5ebb3f762d1a58aaff7f643d7410b&units=metric')
			.then((res) => res.json()).then((res) => { data = res }).catch((err) => {throw err;});
		this.setState({ loading: false, success: true, data: data });
	}
}