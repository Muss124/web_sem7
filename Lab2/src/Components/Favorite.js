import React from 'react';
import { FavoriteSearch } from "./FavoriteSearch.js"
import { connect } from 'react-redux'
import { FavoriteItem } from "./FavoriteItem.js"

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                {
                    this.props.data.map((favCity, i = 0) => {
                        i++;
                        return ([
                            <div class="FavoriteContainer" key={i}>
                                <FavoriteItem value={favCity}  />
                            </div>
                        ])
                    })
                }

            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);