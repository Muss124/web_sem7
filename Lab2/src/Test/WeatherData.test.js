import React from 'react';
import { WeatherData } from '../Components/WeatherData';
import renderer from 'react-test-renderer';

describe('Testing render WeatherData', function () {
  it("Correct input", () => {
    const weatherData = { "coord": { "lon": 37.62, "lat": 55.75 }, "weather": [{ "id": 502, "main": "Rain", "description": "heavy intensity rain", "icon": "10n" }], "base": "stations", "main": { "temp": 1.45, "feels_like": -2.41, "temp_min": 0.56, "temp_max": 2.78, "pressure": 1012, "humidity": 100 }, "visibility": 8000, "wind": { "speed": 3, "deg": 210 }, "rain": { "1h": 8.89 }, "clouds": { "all": 90 }, "dt": 1576589135, "sys": { "type": 1, "id": 9029, "country": "RU", "sunrise": 1576562073, "sunset": 1576587379 }, "timezone": 10800, "id": 524901, "name": "Moscow", "cod": 200 };
    const tree = renderer.create(<WeatherData value={weatherData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});