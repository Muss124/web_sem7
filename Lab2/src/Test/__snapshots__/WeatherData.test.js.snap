// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render WeatherData Favorite weather 1`] = `
<div
  className="WeatherData"
>
  <div
    className="WeatherTop"
  >
    <h2>
      Kiev
    </h2>
    <img
      alt=""
      src="http://openweathermap.org/img/wn/09n.png"
    />
    <h1>
      2.98
       ℃
    </h1>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type="submit"
    >
      ✖
    </button>
  </div>
  <div
    className="WeatherMain"
  >
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Weather
      </div>
      <div
        className="WeatherValue"
      >
        light intensity drizzle
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Humidity (%)
      </div>
      <div
        className="WeatherValue"
      >
        93
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Pressure (hPa)
      </div>
      <div
        className="WeatherValue"
      >
        1014
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Wind speed (meter/sec)
      </div>
      <div
        className="WeatherValue"
      >
        4
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Longitude 
      </div>
      <div
        className="WeatherValue"
      >
        30.52
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Latitude
      </div>
      <div
        className="WeatherValue"
      >
        50.43
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing render WeatherData Header weather 1`] = `
<div
  className="WeatherData"
>
  <div
    className="WeatherTop"
  >
    <h2>
      Kiev
    </h2>
    <img
      alt=""
      src="http://openweathermap.org/img/wn/09n.png"
    />
    <h1>
      2.98
       ℃
    </h1>
  </div>
  <div
    className="WeatherMain"
  >
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Weather
      </div>
      <div
        className="WeatherValue"
      >
        light intensity drizzle
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Humidity (%)
      </div>
      <div
        className="WeatherValue"
      >
        93
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Pressure (hPa)
      </div>
      <div
        className="WeatherValue"
      >
        1014
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Wind speed (meter/sec)
      </div>
      <div
        className="WeatherValue"
      >
        4
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Longitude 
      </div>
      <div
        className="WeatherValue"
      >
        30.52
      </div>
    </div>
    <div
      className="WeatherMainEl"
    >
      <div
        className="WeatherParam"
      >
        Latitude
      </div>
      <div
        className="WeatherValue"
      >
        50.43
      </div>
    </div>
  </div>
</div>
`;
